### PR TITLE
Adding 3.28 release blog post

### PR DESCRIPTION
**Pull request by christinaforney**

### CODE REVIEW

File: blogposts/2021/release-3.28.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+description: "Sourcegraph 3.28 release"
+tags: [blog, release]
+slug: "release/3.28"
+published: true
+changelogItems:
+  # Added
+  - description: "Added `select:commit.diff.added` and `select:commit.diff.removed` for `type:diff` search queries. These selectors return commit diffs only if a pattern matches in `added` (respectively, `removed`) lines."
+    url: https://github.com/sourcegraph/sourcegraph/pull/20328
+    category: Search
+  - description: "Additional language autocompletions for the `lang:` filter in the search bar."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:13]: @rvantonder - what languages and/or how many languages did we add? I'd like to mention them here.

File: blogposts/2021/release-3.28.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20535
+    category: Search
+  - description: "Steps in batch specs can now have an `if:` attribute to enable conditional execution of different steps."
+    url: https://github.com/sourcegraph/sourcegraph/pull/20701
+    category: Batch Changes
+  - description: "Extensions can now log messages through `sourcegraph.app.log` to aid debugging user issues."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:19]: @christinaforney - I recommend cutting this update.

File: blogposts/2021/release-3.28.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20474
+    category:
+  # Changed
+  - description: "User and site credentials used in Batch Changes are now encrypted in the database if encryption is enabled with the `encryption.keys` config."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:23]: @christinaforney - is this worth keeping?

File: blogposts/2021/release-3.28.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/issues/19570
+    category: Batch Changes
+  - description: "All Sourcegraph images within [deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph) now specify the registry. Thanks! @k24dizzle #2901."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:26]: @christinaforney - I recommend cutting.

File: blogposts/2021/release-3.28.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    url: "https://github.com/sourcegraph/deploy-sourcegraph/pull/2901"
+    category:
+  - description: Default reviewers are now added to Bitbucket Server PRs opened by Batch Changes.
+    url: https://github.com/sourcegraph/sourcegraph/pull/20551
+    category: Batch Changes
+  - description: Only site admins can now list users on an instance.
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:32]: @christinaforney - what does this mean exactly?

File: blogposts/2021/release-3.28.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20619
+    category: Admin
+  - description: "Repository permissions can now be enabled for site admins via the `authz.enforceForSiteAdmins` setting."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:35]: @christinaforney - Can you provide context on all of the admin updates. What is the implication for users? Is it worth mentioning?

File: blogposts/2021/release-3.28.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20674
+    category: Admin
+  - description: Site admins can no longer view user added code host configuration.
+    url: https://github.com/sourcegraph/sourcegraph/pull/20851
+    category: Admin
+  - description: Site admins cannot add access tokens for any user by default.
+    url: https://github.com/sourcegraph/sourcegraph/pull/20988
+    category: Admin
+  # Fixed
+  - description: "Indexed search failed when the `master` branch needed indexing but was not the default."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:45]: @christinaforney - I recommend cutting.

File: blogposts/2021/release-3.28.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20260
+    category: Search
+  - description: "`repo:contains(...)` built-in did not respect parameters that affect repo filtering (e.g., `repogroup`, `fork`). It now respects these."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:48]: @christinaforney - recommend cutting.

File: blogposts/2021/release-3.28.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20339
+    category: Repositories
+  - description: "An issue where duplicate results would render for certain `or`-expressions."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:51]: @christinaforney - recommend cutting

File: blogposts/2021/release-3.28.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20480
+    category:
+  - description: "Issue where the search query bar suggests that some `lang` values are not valid."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:54]: @christinaforney  - recommend cutting

File: blogposts/2021/release-3.28.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20534
+    category: Search
+  - description: Pull request event webhooks received from GitHub with unexpected actions no longer cause panics.
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:57]: @christinaforney  - recommend cutting

File: blogposts/2021/release-3.28.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/20571
+    category:
+  - description: "Repository search patterns like `^repo/(prefix-suffix|prefix)$` now correctly match both `repo/prefix-suffix` and `repo/prefix`."
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:60]: @christinaforney - thoughts on including this?

File: blogposts/2021/release-3.28.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/issues/20389
+    category: Search
+  - description: Ephemeral storage requests and limits now match the default cache size to avoid Symbols pods being evicted. The symbols pod now requires 10GB of ephemeral space as a minimum to scheduled.
```
Comment (abschu) [inline blogposts/2021/release-3.28.md:63]: @christinaforney  - thoughts on including this?

**Review by abschu — COMMENTED** [review]
@christinaforney - feedback on the release post.

File: blogposts/2021/release-3.28.md (hunk 1 above)
Comment (rvantonder) [reply blogposts/2021/release-3.28.md:13]: Added 

```
"Assembly",
"Bash",
"Erlang",
"Elixir",
"Dart",
"Haskell",
"Kotlin",
"Julia",
"Objective-C",
"OCaml",
"Scala",
"SQL",
"VBA",
"XML",
"Zig",
```

Full list https://sourcegraph.com/github.com/sourcegraph/sourcegraph@75656d82c51cd628d11ddde15e0dbccd2a093766/-/blob/client/shared/src/search/query/filters.ts#L143-178

File: blogposts/2021/release-3.28.md (hunk 3 above)
Comment (christinaforney) [reply blogposts/2021/release-3.28.md:27]: I don't think it's worth including this at this level.

File: blogposts/2021/release-3.28.md (hunk 13 above)
Comment (christinaforney) [reply blogposts/2021/release-3.28.md:39]: I don't think this should be included.

File: blogposts/2021/release-3.28.md (hunk 12 above)
Comment (christinaforney) [reply blogposts/2021/release-3.28.md:36]: This is a bug fix, I don't think it's worth calling out here.

File: blogposts/2021/release-3.28.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    category: Admin
+  - description: "Repository search patterns like `^repo/(prefix-suffix|prefix)$` now correctly match both `repo/prefix-suffix` and `repo/prefix`."
+    url: https://github.com/sourcegraph/sourcegraph/issues/20389
+    category: Search
+  - description: Ephemeral storage requests and limits now match the default cache size to avoid Symbols pods being evicted. The symbols pod now requires 10GB of ephemeral space as a minimum to scheduled.
+    url: "https://github.com/sourcegraph/deploy-sourcegraph/pull/2369"
+    category:
+---
+
+Sourcegraph 3.28 is now available!
```
Comment (christinaforney) [inline blogposts/2021/release-3.28.md:44]: Are we going to include more details about the post and have a highlight section here? Maybe this is what was waiting on me 🙂 .

File: blogposts/2021/release-3.28.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+slug: "release/3.28"
+published: true
+changelogItems:
+  - description: "Added `select:commit.diff.added` and `select:commit.diff.removed` for `type:diff` search queries. These selectors return commit diffs only if a pattern matches in `added` (respectively, `removed`) lines."
+    url: https://github.com/sourcegraph/sourcegraph/pull/20328
+    category: Search
+  - description: "Bulk comments on many changesets are now available in Batch Changes."
+    url: https://github.com/sourcegraph/sourcegraph/pull/20361
+    category: Batch Changes
+  - description: "The default memory requirements for the `redis-*` containers have been raised by 1GB (to a new total of 7GB). This change allows Redis to properly run its key-eviction routines (when under memory pressure) without getting killed by the host machine. This affects both the docker-compose and Kubernetes deployments."
```
Comment (christinaforney) [inline blogposts/2021/release-3.28.md:15]: I don't think this is a useful update to share.

**Review by christinaforney — COMMENTED** [review]
Added comments on what to include. Am happy to review copy/messaging as well - let me know when ready!

**Comment by rebeccadee** [comment]
@abschu @christinaforney Please request a review when all comments have been resolved! I'm happy to do a final copy edit and publish but prefer to step in when no other changes are expected to avoid making changes that may become redundant 😃  Thanks!

**Pull request merged**